tests/api.c

```diff
@@ -5097,6 +5097,106 @@ static int test_wolfSSL_CertRsaPss(void)
 }
 #endif
 
+
+/* Test RSA-PSS digital signature creation and verification */
+static int test_wc_RsaPSS_DigitalSignVerify(void)
+{
+    EXPECT_DECLS;
+
+/* Early FIPS did not support PSS. */
+#if !defined(HAVE_FIPS) || (defined(HAVE_FIPS_VERSION) && \
+    (HAVE_FIPS_VERSION > 2))
+#if !defined(HAVE_SELFTEST) || (defined(HAVE_SELFTEST_VERSION) && \
+    (HAVE_SELFTEST_VERSION > 2))
+
+#if !defined(NO_RSA) && defined(WC_RSA_PSS) && defined(OPENSSL_EXTRA) && \
+    defined(WOLFSSL_KEY_GEN) && defined(WC_RSA_NO_PADDING)
+    /* Test digest */
+    const unsigned char test_digest[32] = {
+        0x08, 0x09, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05,
+        0x06, 0x07, 0x08, 0x09, 0x00, 0x01, 0x02, 0x03,
+        0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x00, 0x01,
+        0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09
+    };
+    const unsigned int digest_len = 32;
+
+    /* Variables for RSA key generation and signature operations */
+    EVP_PKEY_CTX *pkctx = NULL;
+    EVP_PKEY *pkey = NULL;
+    EVP_PKEY_CTX *sign_ctx = NULL;
+    EVP_PKEY_CTX *verify_ctx = NULL;
+    unsigned char signature[384] = {0};
+    size_t signature_len = sizeof(signature);
+    int modulus_bits = 2048;
+
+    /* Generate RSA key pair to avoid file dependencies */
+    ExpectNotNull(pkctx = EVP_PKEY_CTX_new_id(EVP_PKEY_RSA, NULL));
+    ExpectIntEQ(EVP_PKEY_keygen_init(pkctx), 1);
+    ExpectIntEQ(EVP_PKEY_CTX_set_rsa_keygen_bits(pkctx, modulus_bits), 1);
+    ExpectIntEQ(EVP_PKEY_keygen(pkctx, &pkey), 1);
+
+    /* Create signing context */
+    ExpectNotNull(sign_ctx = EVP_PKEY_CTX_new(pkey, NULL));
+    ExpectIntEQ(EVP_PKEY_sign_init(sign_ctx), 1);
+
+    /* Configure RSA-PSS parameters for signing */
+    ExpectIntEQ(EVP_PKEY_CTX_set_rsa_padding(sign_ctx, RSA_PKCS1_PSS_PADDING),
+                1);
+    ExpectIntGT(EVP_PKEY_CTX_set_rsa_pss_saltlen(sign_ctx, 32), 0);
```

Comment:
Why 32 byte salt length? Maybe a comment?

Reply:
I'll change to 20.  According to our blog, its a good default length.